Source/Core/DolphinQt/Config/Mapping/MappingButton.cpp

```diff
@@ -107,7 +107,7 @@ void MappingButton::Detect()
   if (m_parent->GetDevice() == nullptr || !m_reference->IsInput())
     return;
 
-  installEventFilter(BlockUserInputFilter::Instance());
+  installEventFilter(this);
```

Comment:
It looks like mouse events aren't going to be ignored during detection if you do this.

Reply:
Mouse events don't seem to be ignored when using BlockUserInputFilter, as in mouse movements will be detected and bound, and so will right and left click. Are mouse events intended to be ignored?

Using middle click with BlockUserInputFilter actually detects the event, binds the action (Click 2) and then unbinds it as the detected middle click clears it.

By checking for MouseButtonRelease in MappingButton's eventFilter, it fixes the issue with middle click unbinding itself. One issue with this is that using left click on the button whilst a key is waiting to be bound will not bind left click, it will instead appear to do nothing, but left click can be bound if the user left clicks outside the button.

Checking for MouseButtonPress as well in MappingButton's eventFilter causes single clicks after binding that click to be ignored, meaning if the user were to bind left click, then try single left click another button, nothing would happen. A double click would be required to get the button into 'bind' mode.